lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -0,0 +1,36 @@
+require "pry"
+
+module RuboCop
+  module Cop
+    module RootCops
+      class FactoryFileName < Cop
+        FILE_NAME_ERROR = "Factory name should match file name (one factory per file)".freeze
+        FACTORY = %i[factory].freeze
+
+        def investigate(processed_source)
+          file_path = processed_source.buffer.name
+          @base_name = File.basename(file_path, ".rb")
```

Comment:
I might call this `@base_file_name` to just be a little more clear, but its not a huge difference.